swarm/src/connection.rs

```diff
@@ -110,8 +120,9 @@ where
             max_negotiating_inbound_streams,
         );
         Connection {
-            muxing: Muxing::new(muxer),
+            muxing: muxer,
             handler: wrapped_handler,
+            open_info: VecDeque::with_capacity(8),
```

Comment:
This replaces a `SmallVec` from `Muxing` to ensure that we use the `OpenInfo` in a FIFO manner. A capacity of 8 at least makes only one allocation and thus, the (unbenchmarked) performance profile should be similar.

Open to better ideas.